.github/workflows/check.yml

```diff
@@ -15,30 +14,48 @@ on:
 jobs:
   check:
     # The type of runner that the job will run on
-    runs-on: ubuntu-20.04
+    runs-on: ubuntu-latest
```

Comment:
is it better to use latest? if they push breaking changes it would break our builds? Should we be in control of versions?

Reply:
Seems ubuntu-22.04 is available (latest is 20.04) so could use that, so fixed for the foreseeable future?

I was gonna open a separate issue to sort clippy suggestions, so that this PR remains slim.